tests/test_datediff_level.py

```diff
@@ -164,7 +164,7 @@ def test_datediff_error_logger(test_helpers, dialect):
         cl.datediff_at_thresholds("dob", [1], [])
 
 
-@mark_with_dialects_excluding("sqlite")
+@mark_with_dialects_excluding("sqlite", "postgres")
```

Comment:
It seems a shame to exclude this backend from the portion of the test it works for, but then again this test is fairly large and doing a couple of things, so maybe it makes sense to have a separate PR breaking this up a bit.

Reply:
Yea I had some issues with postgres initially, which I believed which was down to the lack of regex support currently, but may well be wrong. If you have some time to have a look in a separate PR that would be ace.